CRM/Admin/Form/PaymentProcessor.php

```diff
@@ -400,4 +401,34 @@ public function updatePaymentProcessor(&$values, $domainID, $test) {
     civicrm_api3('PaymentProcessor', 'create', $params);
   }
 
+  /**
+   * Check if there is a record with the same name in the db.
+   *
+   * @param string $value
+   *   The value of the field we are checking.
+   * @param array $options
+   *   The daoName and fieldName (optional ).
+   *
+   * @return bool
+   *   true if object exists
+   */
+  public static function paymentProcessorNameExists($value, $options) {
+    $fieldName = 'name';
+    $daoName = CRM_Utils_Array::value(0, $options);
```

Comment:
How does that compare to just having a name-exists function? I mean it should be appropriately domain sensitive but the calling code maybe doesn't need to know that. NB - pass in the entity name rather than the DAO name if you add that - let us dream about a day when DAOs no longer exist :-)

I think the abstract concept of APIs is that they are for externals - but I think CiviCRM is not alone in looking at the logic "Wouldn't it be more robust if everything interacted in the same way & wouldn't it make it easier to swap out internals (e.g replace the PEAR DB layer)  if we had less different ways of interacting with it.